### PR TITLE
AgentMacroReference Event Type

### PR DESCRIPTION
**Pull request by matthewtckr**
Issue #126

### CODE REVIEW

File: src/main/java/org/zendesk/client/v2/model/events/AgentMacroReferenceEvent.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import org.zendesk.client.v2.model.Via;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class AgentMacroReferenceEvent extends Event {
+
+    private static final long serialVersionUID = 1L;
+
+    private Long macroId;
```
Comment (matthewtckr) [inline src/main/java/org/zendesk/client/v2/model/events/AgentMacroReferenceEvent.java]: I'm hesitant to set this as a String (which is what is used in the JSON), I'll submit a few Zendesk Support cases tomorrow for some documentation from Zendesk.

**Comment by jchambers** [comment]
:metal:

This just started making life difficult for us, too.

**Comment by jchambers** [comment]
One other thought: would it make sense to have a `defaultImpl` in the `JsonTypeInfo` annotation in `Event.java`? That way, we can fall back to a not-exactly-desirable-but-at-least-not-catastrophic `UnknownEvent` if new event types get introduced upstream.

**Comment by matthewtckr** [comment]
@jchambers Submitted PR #128 to handle these new, unknown event types.  Feel free to take a look and provide any feedback there.

I agree that `<sarcasm>it's extremely satisfying</sarcasm>` to get notifications like this after-hours because of a Zendesk change.

**Comment by rodrigopalhares** [comment]
Nice job!!! Thx!

**Pull request merged**